engine/servermaster/jobmanager.go

```diff
@@ -252,6 +256,17 @@ func (jm *JobManagerImplV2) SubmitJob(ctx context.Context, req *pb.SubmitJobRequ
 		return resp
 	}
 
+	// Triky way here to set the worker(jobmaster)'s project info
+	defaultMaster, ok := jm.BaseMaster.(*lib.DefaultBaseMaster)
```

Comment:
PTAL~

Reply:
This is not thread-safe.
Maybe we can record a `jobMasterID` to `projectInfo` map in jobmanager when creating worker. (But we need some GC mechanism for the cache). Then in the create worker function we can get `projectInfo` by `jobMasterID`

Reply:
I have lock in  `SetWorkerProjectInfo()`.
`Jobmanager` is the implement of `JobMasterImpl`. It can't modify the implement of BaseJobMaster.CreaterWorker().

Reply:
Solved in latest pr.